src/basic/util.h

```diff
@@ -101,17 +101,23 @@ static inline void qsort_safe(void *base, size_t nmemb, size_t size, comparison_
 #define typesafe_qsort(p, n, func)                                      \
         ({                                                              \
                 int (*_func_)(const typeof(p[0])*, const typeof(p[0])*) = func; \
-                qsort_safe((p), (n), sizeof((p)[0]), (__compar_fn_t) _func_); \
+                qsort_safe((p), (n), sizeof((p)[0]), (comparison_fn_t) _func_); \
         })
 
-static inline void qsort_r_safe(void *base, size_t nmemb, size_t size, int (*compar)(const void*, const void*, void*), void *userdata) {
+static inline void qsort_r_safe(void *base, size_t nmemb, size_t size, __compar_d_fn_t compar, void *userdata) {
```

Comment:
Hm, why use `__compar_d_fn_t` and `comparison_fn_t` in the other case?

Reply:
Indeed. That seems not consistent. I will update them to use `__compar_fn_t`.

At the time I create this PR, I thought that the definition of `typedef_qsort()` is not consistent to the definition of `qsort_safe()`, so I changed it to `comparison_fn_t`. But I may need to change `qsort_safe()` to use `__compar_fn_t`.